contrib/boot-contracts-stateful-prop-tests/tests/pox-4/err_Commands.ts

```diff
@@ -0,0 +1,2283 @@
+import {
+  currentCycle,
+  currentCycleFirstBlock,
+  FIRST_BURNCHAIN_BLOCK_HEIGHT,
+  nextCycleFirstBlock,
+  REWARD_CYCLE_LENGTH,
+} from "./pox_Commands";
+import { DelegateStackExtendCommand_Err } from "./pox_DelegateStackExtendCommand_Err";
+import { DelegateStackIncreaseCommand_Err } from "./pox_DelegateStackIncreaseCommand_Err";
+import { DelegateStackStxCommand_Err } from "./pox_DelegateStackStxCommand_Err";
+import { DelegateStxCommand_Err } from "./pox_DelegateStxCommand_Err";
+import { RevokeDelegateStxCommand_Err } from "./pox_RevokeDelegateStxCommand_Err";
+import { StackAggregationCommitAuthCommand_Err } from "./pox_StackAggregationCommitAuthCommand_Err";
+import { StackAggregationCommitIndexedAuthCommand_Err } from "./pox_StackAggregationCommitIndexedAuthCommand_Err";
+import { StackAggregationCommitIndexedSigCommand_Err } from "./pox_StackAggregationCommitIndexedSigCommand_Err";
+import { StackAggregationCommitSigCommand_Err } from "./pox_StackAggregationCommitSigCommand_Err";
+import { StackAggregationIncreaseCommand_Err } from "./pox_StackAggregationIncreaseCommand_Err";
+import { StackExtendAuthCommand_Err } from "./pox_StackExtendAuthCommand_Err";
+import { StackExtendSigCommand_Err } from "./pox_StackExtendSigCommand_Err";
+import { StackIncreaseAuthCommand_Err } from "./pox_StackIncreaseAuthCommand_Err";
+import { StackIncreaseSigCommand_Err } from "./pox_StackIncreaseSigCommand_Err";
+import { StackStxAuthCommand_Err } from "./pox_StackStxAuthCommand_Err";
+import { StackStxSigCommand_Err } from "./pox_StackStxSigCommand_Err";
+import { DisallowContractCallerCommand_Err } from "./pox_DisallowContractCallerCommand_Err";
+
+const POX_4_ERRORS = {
+  ERR_STACKING_INSUFFICIENT_FUNDS: 1,
+  ERR_STACKING_INVALID_LOCK_PERIOD: 2,
+  ERR_STACKING_ALREADY_STACKED: 3,
+  ERR_STACKING_NO_SUCH_PRINCIPAL: 4,
+  ERR_STACKING_PERMISSION_DENIED: 9,
+  ERR_STACKING_THRESHOLD_NOT_MET: 11,
+  ERR_STACKING_INVALID_AMOUNT: 18,
+  ERR_STACKING_ALREADY_DELEGATED: 20,
+  ERR_DELEGATION_TOO_MUCH_LOCKED: 22,
+  ERR_STACK_EXTEND_NOT_LOCKED: 26,
+  ERR_STACKING_IS_DELEGATED: 30,
+  ERR_STACKING_NOT_DELEGATED: 31,
+  ERR_DELEGATION_ALREADY_REVOKED: 34,
+};
+
+export function ErrCommands(
+  wallets: Map<StxAddress, Wallet>,
+  stackers: Map<StxAddress, Stacker>,
+  network: Simnet,
+): fc.Arbitrary<PoxCommand>[] {
+  const cmds = [
+    // StackStxAuthCommand_Err_Stacking_Already_Stacked_1
+    fc.record({
+      wallet: fc.constantFrom(...wallets.values()),
+      authId: fc.nat(),
+      period: fc.integer({ min: 1, max: 12 }),
+      margin: fc.integer({ min: 1, max: 9 }),
+    }).map((
+      r: {
+        wallet: Wallet;
+        authId: number;
+        period: number;
+        margin: number;
+      },
+    ) =>
+      new StackStxAuthCommand_Err(
+        r.wallet,
+        r.authId,
+        r.period,
+        r.margin,
+        function (
+          this: StackStxAuthCommand_Err,
+          model: Readonly<Stub>,
+        ): boolean {
```

Comment:
It's not needed to specify the type in the case.
Typescript can infer the types and `StackStxAuthCommand_Err` expects this signature.

Overall, it could slightly simplify all of the code in this file